core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -68,7 +68,7 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
         if (id.fetcherId >= newSize)
           thread.shutdown()
         partitionStates.forKeyValue { (topicPartition, currentFetchState) =>
-            val initialFetchState = InitialFetchState(currentFetchState.topicId, thread.sourceBroker,
+            val initialFetchState = InitialFetchState(currentFetchState.topicId, thread.leader.brokerEndPoint(),
```

Comment:
nit: I wonder if we could omit the parenthesis when calling `brokerEndPoint`. There are two other cases below.